frontend/awx/administration/instances/hooks/useSelectAssociatePeers.tsx

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { MultiSelectDialog, usePageDialog } from '../../../../../framework';
+import { usePeersTabFilters } from '../Instances';
+import { awxAPI } from '../../../common/api/awx-utils';
+import { useAwxView } from '../../../common/useAwxView';
+import { useAssociatePeersToInstance } from './useAssociatePeersToInstance';
+import { usePeersToAssociateColumns } from './usePeersColumns';
+import { useGetItem } from '../../../../common/crud/useGet';
+import { useParams } from 'react-router-dom';
+import { Peer } from '../../../interfaces/Instance';
+
+function SelectAssociates(props: {
+  accessUrl?: string;
+  title: string;
+  onSelect: (peers: Peer[]) => void;
+  confirmText?: string;
+}) {
+  const { t } = useTranslation();
+  const params = useParams<{ id?: string }>();
+  const { data: instance } = useGetItem<Peer>(awxAPI`/instances/`, params.id);
```

Comment:
Should this be Instance instead of Peer?

Reply:
since `Peer`  extends from `Instance` it works both ways but for clarity purposes i agree it should be `Instance`